src/battle_script_commands.c

```diff
@@ -6157,6 +6157,21 @@ static void Cmd_switchindataupdate(void)
     for (i = 0; i < sizeof(struct BattlePokemon); i++)
         monData[i] = gBattleResources->bufferB[battler][4 + i];
 
+    // Edge case: the sent out pokemon has 0 HP. This should never happen.
+    if (gBattleMons[battler].hp == 0)
+    {
+        // If it's a test, mark it as invalid.
+        if (gTestRunnerEnabled)
+        {
+            TestRunner_Battle_InvalidNoHPMon(battler, gBattlerPartyIndexes[battler]);
+        }
+        // TODO: Handle in-game scenario.
```

Comment:
Do you have an idea of what should happen in this situation in-game?
If not, I think combining the gTestRunnerEnabled and HP checks should suffice.

Reply:
I made it so the game looks for a valid replacement. Though, this should never happen anyway(unless cheats or bugs).